analyzers/tests/SonarAnalyzer.UnitTest/TestCases/BooleanLiteralUnnecessary.cs

```diff
@@ -222,4 +222,13 @@ public class Socket
     {
         public bool? IsStateValid { get; set; }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/7999
+    class Repro7999CodeFixError
+    {
+        void Method(bool cond)
+        {
+            if (cond == false) { } // Noncompliant, TP but code fix is wrong - it should be fixed to "if (!cond)"
```

Comment:
Any better idea to mark that the issue lies with the fix and not really with the rule?

Reply:
I think you can copy (or even move) the explanation to the code fix, since that is where the mistake happens.

Reply:
That's not really an option due to how the fixes are validated. :/